### DISCUSSION TITLE
credential required for basic auth Error while Using --username --password flags to helm install for private distribution repo

### DISCUSSION THREAD

**Discussion by rahulbapumore** [distribution/distribution#4493] [Q&A]
We are trying to install helm chart stored at private distribution registry, but we are getting error while installing even though we pass --username , --password flag to helm install command.
we want authentication to happen with helm install command only, we dont want to login to repo beforehand
Below is the command, let me know if I am missing anything in below command - 

` helm install rel oci://abc.io/chart1 --username=admin --password=admin   --ca-file ca.crt --cert-file client.cert  --key-file client.key
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /src/admin.conf
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /src/admin.conf
Error: INSTALLATION FAILED: GET "https://abc.io/v2/chart1/tags/list": credential required for basic auth
`

Thanks

**Comment by rahulbapumore** [comment]
@milosgajdos Any idea on query?

**Reply by milosgajdos** [reply]
Please stop @-ing me on your queries. There are 10+ other maintainers on this repo. Furthermore, this is a discussion forum, be patient and wait until other users chip 

I'll give you a hint:
* can you login into registry using helm at all? https://helm.sh/docs/helm/helm_registry_login/ I would start here before attempting to install the chart

I would also strongly encourage you to change permissions on your `admin.conf` file:
```
chmod 600 /src/admin.conf
```

**Comment by rahulbapumore** [comment]
Hi @milosgajdos 
Sorry for tagging directly.
We tried by giving permission to /src/admin.conf, but still it doesnt accept username and password while installing helm chart.
After doing helm registry login it works, but we want to pass username and password to helm install/upgrade command without doing helm registry login, doing so it gives **error credential required for basic auth**

**Reply by milosgajdos** [reply]
Look, I don't know why helm requires you to log in to the registry first. I'm not a helm developer not a user. My suspicion is it stores the login credentials locally. But that's really a question you should be asking helm maintainers 🤷‍♂️ 

As for pulling images from private registries usually in k8s if you want to access some private registry you need to create an image pull secret. Again, that's a k8s question, not distribution question
